llvm/test/tools/llc/cas_options.ll

```diff
@@ -0,0 +1,14 @@
+; REQUIRES: default_triple
+; RUN: not llc -o /dev/null --filetype=asm --cas-backend --mtriple=arm64-apple-ios13.4.0 \
+; RUN:   2>&1 | FileCheck %s --check-prefix=OUTPUT_TYPE
+; RUN: not llc -o /dev/null --filetype=null --cas-backend --mtriple=arm64-apple-ios13.4.0 \
+; RUN:   2>&1 | FileCheck %s --check-prefix=OUTPUT_TYPE
+; OUTPUT_TYPE: CAS Backend requires .obj output
+
+; RUN: not llc -o /dev/null --filetype=obj --cas-backend --mtriple=i386-pc-linux-gnu \
```

Comment:
If `--filetype=asm --cas-backend` is invalid, should this be a third filetype instead?
(I'm fine with no as an answer, just want to understand the reasoning)

Reply:
To be honest I was blindly following the example of what was done with `llvm-mc`.  Your idea makes sense to me, though it could touch more upstream code. Maybe let's think about this in the future with the two tools together in mind?

Reply:
Yeah, if we touch more code we should do that upstream and not increase the size of the diff unnecessarily

Reply:
In short: LGTM